File: web_app/frontend/src/app/interfaces/musical-band.ts
import { UserProfile } from "./user-profile"
import { BandHQ } from "./bandHQ"

export interface MusicalBand {
    bandId: string
    name: string
    musicGenre?: string
    dateFormed: Date
    isComplete: boolean
    members: UserProfile[]
    hq?: BandHQ
    //creatorUsername: string
}

export interface BandFormData {
    username: string
    formData?: MusicalBand
};

File: web_app/backend/Entities/MusicalBand.cs
using System.Collections.ObjectModel;
using System.ComponentModel.DataAnnotations;

namespace backend.Entities
{
    public class MusicalBand
    {
        [Key]
        public string BandId { get; set; } = null!;
        public virtual BandHQ? HQ { get; set; }

        public string Name { get; set; } = null!;
        public string? MusicGenre { get; set; }
        public DateTime DateFormed { get; set; }
        public bool IsComplete { get; set; }

        //public string CreatorId { get; set; }
        public virtual ICollection<UserProfile> Members { get; set; } = new Collection<UserProfile>();
        public virtual ICollection<BandUserMatch>? Matches { get; set; }
    }
}
